src/manifest/manifestloader.js

```diff
@@ -88,13 +90,15 @@ function getStreamUrl(data) {
 }
 
 export default {
-  load: (mediaUrl, { windowType, initialWallclockTime, onError, onSuccess } = {}) => {
+  load: (mediaUrl, { windowType, initialWallclockTime } = {}) => {
     if (/\.mpd($|\?.*$)/.test(mediaUrl)) {
-      retrieveDashManifest(mediaUrl, { windowType, initialWallclockTime, onError, onSuccess })
-    } else if (/\.m3u8($|\?.*$)/.test(mediaUrl)) {
-      retrieveHLSManifest(mediaUrl, { windowType, initialWallclockTime, onError, onSuccess })
-    } else {
-      onError("Invalid media url")
+      return retrieveDashManifest(mediaUrl, { windowType, initialWallclockTime })
+    }
+
+    if (/\.m3u8($|\?.*$)/.test(mediaUrl)) {
```

Comment:
I know you didn't write this regex, but we could just make the query param an optional match like `/\.m3u8(\?.*)?$/`, I know regex comprehension isn't most people's strong suite in the first place, but I think this reads better since you can interpret it entirely forward (As a bonus, it's ever so slightly faster in the case where there's a query param too, taking one less step).